ui/apps/platform/src/Containers/Collections/CollectionsTable.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { Button, ButtonVariant, Truncate, Bullseye, Text } from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons';
+import { TableComposable, TableVariant, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+
+import EmptyStateTemplate from 'Components/PatternFly/EmptyStateTemplate';
+import LinkShim from 'Components/PatternFly/LinkShim';
+import useTableSelection from 'hooks/useTableSelection';
+import { CollectionResponse } from 'services/CollectionsService';
+import { collectionsPath } from 'routePaths';
+import { collectionsTableColumns } from './collectionsTableColumnDescriptors';
+
+export type CollectionsTableProps = {
+    collections: CollectionResponse[];
+    hasWriteAccess: boolean;
+};
+
+function CollectionsTable({ collections, hasWriteAccess }: CollectionsTableProps) {
+    const history = useHistory();
+    const { selected, allRowsSelected, onSelect, onSelectAll } = useTableSelection(collections);
+
+    function onEditCollection(id: string) {
+        history.push({
+            pathname: `${collectionsPath}/${id}`,
+            search: 'action=edit',
+        });
+    }
+
+    function onCloneCollection(id: string) {
+        history.push({
+            pathname: `${collectionsPath}/${id}`,
+            search: 'action=clone',
+        });
+    }
+
+    return (
+        <>
+            <TableComposable variant={TableVariant.compact}>
+                <Thead>
+                    <Tr>
+                        {hasWriteAccess && (
+                            <Th
+                                select={{
+                                    onSelect: onSelectAll,
+                                    isSelected: allRowsSelected,
+                                }}
+                            />
+                        )}
+                        {collectionsTableColumns.map(({ Header, width }) => (
+                            <Th key={Header} modifier="wrap" width={width}>
+                                {Header}
+                            </Th>
+                        ))}
+                    </Tr>
+                </Thead>
+                <Tbody>
+                    {collections.length === 0 && (
+                        <Tr>
+                            <Td colSpan={8}>
```

Comment:
If I remember correctly, UX team prefers either empty state or table @vjwilson can you comment?

If so, you can lift up `EmptyStateTemplate` to conditional rendering in `CollectionsTablePage`

Reply:
I believe that is correct. Let's just mention that we're doing it in the UX-UI joint meeting.

Reply:
I went with this approach since this table will have a toolbar in a follow up PR that can be used to filter the results. Looking into this more though, this is a good one to bring up in the joint meeting today since behavior is fairly inconsistent in other places:

- It looks like the CVE table in Vuln Mgmt uses the same approach here (EmptyState in place of `<tbody>`)
- Violations and Policies simply leave the table empty, with headers/toolbar intact
- Integrations and Access Control replace the entire table with an EmptyState (these pages do not have a table toolbar though)
- Risk Acceptance is the same as Integrations/Access Control, but _does_ have a toolbar with filtering, which leads to an inconsistent state. I'll file a bug for this one.

- [ ] Create ticket for Risk Acceptance after guidance from UX

Reply:
Created the following related issues after meeting with UX:
https://issues.redhat.com/browse/ROX-12821
https://issues.redhat.com/browse/ROX-12822